src/config.ts

```diff
@@ -188,6 +198,17 @@ export class ConfigurationReader implements vscode.Disposable {
 
   get useCMakeServer(): boolean { return this.configData.useCMakeServer; }
 
+  get cmakeCommunicationMode(): CMakeCommunicationMode {
+    let communicationMode = this.configData.cmakeCommunicationMode;
+    if (this.useCMakeServer) {
+      log.warning(localize(
+          'please.upgrade.configuration',
+          'The setting \'useCMakeServer\' is replaced by \'cmakeCommunicationMode\'. Please upgrade your configuration.'));
+      communicationMode = 'serverAPI';
```

Comment:
If the settings conflict, I would rather take `cmakeCommunicationMode` over this one.  (e.g. if communication mode is fileAPI, we shouldn't downgrade it to server).  I think you should only allow setting it to server here if communicationMode is automatic (the default value).  I believe vscode has an API that would let us know if "automatic" was set in the user config versus being the default, unset value, but we might be able to ignore that for now.